### ISSUE TITLE
hot_mode_dev_cycle_macos_target__benchmark fails on ARM M1 mac

### ISSUE THREAD

**Issue by jmagman** [flutter/flutter#85654]
`hot_mode_dev_cycle_macos_target__benchmark` is still failing on these arm Macs, though it's passing locally for me. 
 There's still something off about that CI environment? 

https://luci-milo.appspot.com/p/flutter/builders/prod/Mac_ios_staging%20hot_mode_dev_cycle_macos_target__benchmark/637
```
[hot_mode_dev_cycle_macos_target__benchmark] [STDOUT] stderr: [+1023 ms] Error waiting for a debug connection: The log reader stopped unexpectedly.
[hot_mode_dev_cycle_macos_target__benchmark] [STDOUT] stderr: [   +2 ms] Error launching application on macOS.
[hot_mode_dev_cycle_macos_target__benchmark] [STDOUT] stdout: [   +2 ms] "flutter run" took 52,330ms.
```

**Comment by jmagman** [comment]
@keyonghan are there any `Runner*.crash` files in ~/Library/Logs/DiagnosticReports for that user` with a timestamp matching up to a failing test?

**Comment by jmagman** [comment]
> are there any `Runner*.crash` files 

Or `flutter_gallery*.crash` or `Flutter Gallery*.crash` files

**Comment by keyonghan** [comment]
These are the crash list in mac5, but no mentioned ones:
[mac5_crash.txt](https://github.com/flutter/flutter/files/6744291/mac5_crash.txt)

**Comment by godofredoc** [comment]
@jmagman any other logs that we can collect to help find the root cause?

**Comment by jmagman** [comment]
Since I couldn't reproduce on my machine, maybe it's something with the Xcode certs?
Is there any way I can get a `sudo sysdiagnose -Q` (the `-Q` is optional, but so it doesn't take so long to run)?  I'm looking for logging related to the sandbox or security.

**Comment by godofredoc** [comment]
Using `sudo gem install cocoapods` makes it work locally however it still fails to run remotely.

**Comment by jmagman** [comment]
> > I collected them from a VNC session. Do you want me to collect another one over an ssh session?

Yes, that would be more accurate.  I suspect the lack of windowserver is the problem.
If the user is logged in vnc, and then you start an ssh session with the same user, do the tests pass?

**Comment by godofredoc** [comment]
It fails consistently for me when running from the ssh terminal. Did you by any chance run it from a terminal inside the vnc session?

**Comment by godofredoc** [comment]
Do we also need to remove the defaults from https://github.com/flutter/flutter/blob/be8d29ffc639d22db81dc7ceadcf4b1423613452/dev/devicelab/lib/framework/ios.dart#L146

I removed the ios_signing dependency but the tests are still failing:

* https://chromium-swarm.appspot.com/task?id=56b28883434a0c10
* https://chromium-swarm.appspot.com/task?id=56b28842c90c7110

**Comment by jmagman** [comment]
Thanks @godofredoc!
